Swap quote preview

tigranpetrossian opened the pull request:
This is work in progress adding inline quote preview to Swap, as well as all the logic necessary for powering both the inline preview and the review screen (upcoming).

* `useLiveEstimate` hook for tracking live quote and fees, managing the refetching across screens. Quote refetch interval is persisted across screens.
* Basic quote preview showing rate, fees, and an unobtrusive countdown via circular progress indicator
* Moving some of the few remaining calculations to Bignumber for accurate precision

+ Some small bugfixes and refactoring here and there.

### Screencast with possible states
https://github.com/user-attachments/assets/90137a33-2d8e-44c4-972a-8089f0684c96

tigranpetrossian committed:
feat(mobile): add live swap estimate hook, simplify swap form layout

tigranpetrossian committed:
fix(mobile): reset swap amount when base asset changes

tigranpetrossian committed:
feat(mobile): add native asset market data query to swap state

tigranpetrossian committed:
feat(mobile): rename swap providerFee -> providerFeePercentage for clarity

tigranpetrossian committed:
feat(mobile): add fee breakdown to live swap estimate hook, fix precision for various fields

tigranpetrossian committed:
feat(mobile): add swap quote inline preview

tigranpetrossian commented on apps/mobile/src/features/swap/components/quote-preview/quote-preview-content.tsx:
```diff
@@ -0,0 +1,90 @@
+    opacity: withTiming(isRefetching ? 0.5 : 1),
+  }));
+
+  const elapsed = lastStartedAt ? Date.now() - lastStartedAt : 0;
```
This will be extracted and re-used in the review screen.

pete-watters commented on apps/mobile/src/features/swap/swap-state/swap.queries.ts:
```diff
@@ -140,7 +148,7 @@ export function useAssetMarketDataQuery({
     },
     enabled: isDefined(asset),
-    refetchInterval: 30000, // Refetch every 30 seconds
+    refetchInterval: 30000,
```
If `staleTime: Infinity` I think that `refetchInterval: 30000` becomes irrelevant and redundant?

Should this never be refetched?

tigranpetrossian replied:
`refetchInterval` works regardless. `staleTime` only affects `refetchOnWindowFocus` type options.

pete-watters reviewed (approved):
Great work @tigranpetrossian ! It's exciting to see this coming together piece by piece alongside many improvements

tigranpetrossian committed:
fix(mobile): use stable secondary target amount

ensures both target amounts are stable while new quote is loading, to support the pulsing animation

tigranpetrossian commented:
Merging to unblock ongoing work. Post-merge feedback is still very welcome.
